### ISSUE TITLE
Updating layer metadata using geonode admin causes error

### ISSUE THREAD

**Issue by NyakudyaA** [kartoza/geonode#127]
# Problem

When a layer's metadata is edited using the geonode admin page it has mandatory fields like : SLD which prevent it from saving the metadata. If you put a fake url for the metadata it saves the record. 

![adi](https://cloud.githubusercontent.com/assets/2510900/21049751/5834a080-be1f-11e6-863a-491fd0b21016.png)

After this change if you try to access the layer in geonode/layers/layer_name it gives an error because it now tries to look for the sld.

![index](https://cloud.githubusercontent.com/assets/2510900/21049750/5830dc98-be1f-11e6-97ca-8ac0940259e7.png)

# Solution
Disable mandatory fields when editing layers through geonode admin page.

**Comment by gubuntu** [comment]
@NyakudyaA pls test again and comment or close

**Comment by NyakudyaA** [comment]
@gubuntu Errors still exist and we need to disable all fields associated with geonode.
* Open a layer in geonode admin and click save.
* An error message appears even though the metadata is fully populated.
![error](https://user-images.githubusercontent.com/2510900/30734667-6fa40cda-9f7c-11e7-8671-d61fb6649b67.png)

**Comment by gubuntu** [comment]
@NyakudyaA @lucernae I wonder if this affects metadata upload as well? If we upload a new metadata doc to replace one for a layer in QGIS server backend, is GeoNode rejecting it because of lack of mandatory geoserver fields?

@lucernae I'm labelling this Ready as it might be an issue for the GeoSAFE demo instance.

@NyakudyaA could you test this and other metadata issues while setting up the demo data?

**Comment by gubuntu** [comment]
related to #251 and #143

**Comment by gubuntu** [comment]
@lucernae is this affected by https://github.com/kartoza/geosafe/pull/378

**Comment by lucernae** [comment]
@gubuntu yes. On several notes.
1. Editing metadata through Admin page is not covered. The admin page is tightly configured for GeoServer. We may be able configure the layout for QGIS Server only if we're able to override it. And yes, it rejects because it was mandatory field for GeoServer layers.
2. Layer Upload Metadata now works. To update metadata, you can download ISO metadata first from the layer. Edit it, and reupload just the metadata on the same layer. Note that since InaSAFE only handles ISO metadata, we didn't cover any other type of metadata (if you download it, it won't have InaSAFE keywords, we don't know where to put it on other types).
3. I put unittests and did some light testing on testing server. If you upload metadata which doesn't have inasafe keywords, the layer will be excluded from GeoSAFE InaSAFE layers. If you include the metadata again, it will be recognized again. Some mandatory field that is recognized by GeoNode, like layer title will also be consumed and change the actual title of the layer.
4. Using Layer Upload Metadata now should not break the whole site. The metadata will not be duplicated, the base file will be safe, and the styles will still be there.

Let me know if you have some problem with it. There maybe some edge cases which I didn't think of.

**Comment by gubuntu** [comment]
I don't think the normal use case is to edit metadata via admin but rather via uploading or the UI so I'll close this. (open again if you disagree @NyakudyaA )

thanks for comments @lucernae

**Comment by NyakudyaA** [comment]
@gubuntu The recommended way of editing metadata is by using the UI which allows you to upload new metadata records but the backend also provides more options in terms of editing metadata. 
* A user can see the full metadata that has been generated and you can edit this.
* There other metadata fields which are exposed in the admin interface which are not available in the UI.

Although it should not be high priority to fix this I suggest we need to fix this at some point so that users can have a seamless workflow.

The only two fields which are required are the `store` and `style`. The easiest way would be just to disable these two since QGIS backend does not use them or alternatively just fill them in with dummy values.

**Comment by gubuntu** [comment]
OK let's leave this in the backlog

**Comment by lucernae** [comment]
Yes, I agree with @NyakudyaA we should be able to disable/hide Geoserver's specific field such as store and style. We can probably override the admin page.
